LibreHardwareMonitorLib/Hardware/Motherboard/Motherboard.cs

```diff
@@ -148,5 +158,32 @@ public void Close()
                     hardware.Close();
             }
         }
+
+        EmbeddedController embeddedController(Model model, ISettings settings)
+        {
+            var sources = new List<ECSource>();
+            switch (model)
+            {
+                case Model.ROG_CROSSHAIR_VIII_HERO:
+                    {
+                        sources.AddRange(new ECSource[]{
+                        new ECSource("Chipset", 0x3A, SensorType.Temperature, EmbeddedController.ReadByte),
+                        new ECSource("CPU", 0x3B, SensorType.Temperature, EmbeddedController.ReadByte),
+                        new ECSource("Motherboard", 0x3C, SensorType.Temperature, EmbeddedController.ReadByte),
+                        new ECSource("T Sensor", 0x3D, SensorType.Temperature, EmbeddedController.ReadByte),
+                        new ECSource("VRM", 0x3E, SensorType.Temperature, EmbeddedController.ReadByte),
+
+                        new ECSource("CPU Opt", 0xB0, SensorType.Fan, EmbeddedController.ReadWordBE),
+                        new ECSource("Chipset", 0xB4, SensorType.Fan, EmbeddedController.ReadWordBE),
+                        new ECSource("Water Flow", 0xBC, SensorType.Fan, EmbeddedController.ReadWordBE), // TODO: how to convert this value to litres per minute? 
+
+                        new ECSource("CPU", 0xF4, SensorType.Current, EmbeddedController.ReadByte)
+                    });
+                        break;
+                    }
```

Comment:
Maybe my VS needs tuning some settings? This is how it wants to format the file, and I want to add that editing the Nct677x constructor code was a bad experience because VS editor was adding a tab level for every block and line and I had to un-tab it manually.

Reply:
Possibly, it should follow the editorconfig usually. Are you running VS2019?

Reply:
Microsoft Visual Studio Community 2019, Version 16.10.0

Reply:
That should work, but I think it's best to manually tab some back for now, I'll take a look at the editorconfig later.

Reply:
OK, thanks!